app/controllers/UserDataController.scala

```diff
@@ -28,18 +31,48 @@ class UserDataController(dynamo: Dynamo, val deps: BaseFaciaControllerComponents
   }
 
   def putFrontIds() = APIAuthAction { request =>
-
-    val frontIds: Option[List[String]] = request.body.asJson.flatMap(jsValue =>
-      jsValue.asOpt[List[String]])
-
-    frontIds match {
-      case Some(ids) => {
-        val userEmail = request.user.email
-        Scanamo.exec(dynamo.client)(userDataTable.update('email -> request.user.email, set('frontIds -> ids)))
+    // We handle both cases here, because of the migration from a list to a map by priority
+    val maybeFrontIds: Option[List[String]] = request.body.asJson.flatMap(
+      _.asOpt[List[String]])
+    val maybeFrontIdsByPriority: Option[Map[String, List[String]]] = request.body.asJson.flatMap(
```

Comment:
It might be clearer to split this into two functions? One for the old path and one for the new?

Reply:
Yeah, I do prefer this. Good point, thanks 👍